presto-main/src/main/java/com/facebook/presto/sql/planner/iterative/rule/PushAggregationThroughOuterJoin.java

```diff
@@ -272,7 +272,7 @@ private MappedAggregationInfo createAggregationOverNull(AggregationNode referenc
             Symbol aggregationSymbol = entry.getKey();
             AggregationNode.Aggregation aggregation = entry.getValue();
             AggregationNode.Aggregation overNullAggregation = new AggregationNode.Aggregation(
-                    (FunctionCall) new ExpressionSymbolInliner(sourcesSymbolMapping::get).rewrite(aggregation.getCall()),
+                    (FunctionCall) inlineSymbols(sourcesSymbolMapping::get, aggregation.getCall()),
```

Comment:
1. If `inlineSymbols` always returns the same type as the input (does it?) we could call it without explicit cast here
2. Can `inlineSymbols` overload take `sourcesSymbolMapping` (without `::get`) ? It would be applicable in a lot of places.

Reply:
1. I thought about this and unfortunately it does not work always. For example when pass SymbolReference then as result you could get mapped Expression.
2. Sure.